auto_nag/scripts/crash_samll_volume.py

```diff
@@ -0,0 +1,188 @@
+
+        top_crash_signatures = [
+            signature
+            for signature in signatures
+            if signature in self.topcrash_signatures
+        ]
+
+        keep_topcrash_startup = any(
+            any(
+                criterion["is_startup"]
+                for criterion in self.topcrash_signatures[signature]
+            )
+            for signature in top_crash_signatures
+        )
+
+        keywords_to_remove = None
+        if not top_crash_signatures:
+            keywords_to_remove = existing_keywords - {"topcrash", "topcrash-startup"}
+        elif not keep_topcrash_startup:
+            keywords_to_remove = existing_keywords - {"topcrash-startup"}
+
+        data[bugid] = {
+            "severity": bug["severity"],
+            "keep_topcrash": bool(top_crash_signatures),
+            "keywords_to_remove": keywords_to_remove,
+            "signatures": signatures,
+        }
+
+        return bug
+
+    def _get_low_volume_crash_signatures(self, bugs):
+        signatures = {
+            signature
+            for bug in bugs.values()
+            if not bug["keep_topcrash"] and bug["severity"] in HIGH_SEVERITY
+            for signature in bug["signatures"]
+        }
+
+        signature_volume = Topcrash().fetch_signature_volume(signatures)
+
+        low_volume_signatures = {
+            signature for signature, volume in signature_volume.items() if volume < 5
+        }
+
+        return low_volume_signatures
+
+    def get_bugs(self, date="today", bug_ids=..., chunk_size=None):
+        bugs = super().get_bugs(date, bug_ids, chunk_size)
+        self.set_autofix(bugs)
+
+        # Keep only bugs with an autofix
+        bugs = {
+            bugid: bug for bugid, bug in bugs.items() if bugid in self.autofix_changes
+        }
+
+        return bugs
+
+    def set_autofix(self, bugs):
+        """Set the autofix for each bug."""
+
+        low_volume_signatures = self._get_low_volume_crash_signatures(bugs)
+
+        for bugid, bug in bugs.items():
+            autofix = {}
+            reasons = []
+            if bug["keywords_to_remove"]:
+                reasons.append(
+                    "The crash signature linked to this bug is not a topcrash signature anymore."
+                    if len(bug["signatures"]) == 1
+                    else "The crash signatures linked to this bug are not in the topcrash signatures anymore."
+                )
+                autofix["keywords"] = {"remove": list(bug["keywords_to_remove"])}
+
+            if bug["severity"] in HIGH_SEVERITY and all(
+                signature in low_volume_signatures
+                or signature in self.blocked_signatures
+                for signature in bug["signatures"]
+            ):
+                reasons.append(
+                    "Since the crash volume is very low, the severity is downgraded to `S3`. "
+                    "Feel free to change it back if you think the bug is still critical."
```

Comment:
We ask here to "change it back" if the developer does not agree. However, with the current implementation, the bot will decrease the severity again to `S3` after 30 days if the bug is still open.

@marco-c wdyt?

Reply:
Maybe we should parse the history and avoid resetting to S3 if it was set again to S2 by anyone that is not the bot. There could be legitimate reasons for crashes with low volume but high severity (e.g. a crash with a feature that is not released yet, and so the few crashes are explained by the low number of users testing the feature).

Reply:
Done in 024de6c82a9b811301990b4e5fa5d52192b9aab3. With this, the bot does not downgrade the severity more than once.